dsl_parser/tests/test_import_resolver.py

```diff
@@ -0,0 +1,206 @@
+########
+# Copyright (c) 2015 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+import testtools
+
+from dsl_parser import utils
+from dsl_parser.constants import RESOLVER_IMPLEMENTATION_KEY, \
+    RESLOVER_PARAMETERS_KEY
+from dsl_parser.import_resolver.abstract_import_resolver import \
+    AbstractImportResolver
+from dsl_parser.import_resolver.default_import_resolver import \
+    DefaultResolverValidationException, DefaultImportResolver, \
+    DEFAULT_RESLOVER_RULES_KEY
+
+default_resolver_class_path = "%s:%s" % (
+    DefaultImportResolver.__module__, DefaultImportResolver.__name__)
+
+
+class MockCustomImportResolverException(Exception):
+    pass
+
+
+class CustomImportResolver(AbstractImportResolver):
+    def __init__(self, custom_reolver_parameters):
+        self.parameters = custom_reolver_parameters
+
+    def resolve(self, import_url):
+        pass
+
+custom_resolver_class_path = "%s:%s" % (
+    CustomImportResolver.__module__, CustomImportResolver.__name__)
+
+
+class CustomImportResolverWithoutInit(AbstractImportResolver):
+    def resolve(self, import_url):
+        pass
+custom_no_init_resolver_class_path = "%s:%s" % (
+    CustomImportResolverWithoutInit.__module__,
+    CustomImportResolverWithoutInit.__name__)
+
+
+class FailedToInitializeCustomImportResolver(AbstractImportResolver):
+    def __init__(self):
+        raise MockCustomImportResolverException('mock exception')
+
+    def resolve(self, import_url):
+        pass
+
+failed_custom_resolver_class_path = "%s:%s" % (
+    FailedToInitializeCustomImportResolver.__module__,
+    FailedToInitializeCustomImportResolver.__name__)
+
+
+class CreateImportResolverTests(testtools.TestCase):
+
+    def _test_create_import_resolver(self,
+                                     resolver_configuration=None,
+                                     expected_resolver=None,
+                                     expected_params_name=None,
+                                     err_msg_regex=None):
+        if expected_resolver:
+            resolver = utils.create_import_resolver(resolver_configuration)
+            self.assertEqual(resolver.__class__, expected_resolver.__class__)
+            if expected_params_name:
+                self.assertEqual(
```

Comment:
I don't think there's a test the verifies this assert (meaning that the params don't match)